sdk/src/main/scala/com/horizen/account/state/AccountStateView.scala

```diff
@@ -298,9 +304,16 @@ class AccountStateView(
 
   override def getConsensusEpochNumber: Option[ConsensusEpochNumber] = metadataStorageView.getConsensusEpochNumber
 
-  override def getFeePayments(withdrawalEpoch: Int): Seq[BlockFeeInfo] =
-    metadataStorageView.getFeePayments(withdrawalEpoch)
+  override def getFeePayments(withdrawalEpoch: Int, blockToAppendFeeInfo: Option[AccountBlockFeeInfo] = None): Seq[AccountBlockFeeInfo] = {
```

Comment:
Why do we return the seq of `sources` instead of `AccountFeePaymentsUtils.getForgersRewards`, so the resulting fee payments?
I assume, no one except the state inside is interested in the list of `AccountBlockFeeInfo`?

The root of fee payments expected to be the root of pairs `(AddressProposition, BigInteger)`. Otherwise, to understand the changes and verify the AccountBlock.feePaymentsHash a third-party need to know how to calculate `getForgersRewards`.